xskillscore/core/np_deterministic.py

```diff
@@ -188,6 +189,55 @@ def _pearson_r(a, b, weights, axis, skipna):
     return res
 
 
+def _r2(a, b, weights, axis, skipna):
+    """
+    ndarray implementation of sklearn.metrics.r2_score.
+
+    Parameters
+    ----------
+    a : ndarray
+        Input array.
+    b : ndarray
+        Input array.
+    axis : int
+        The axis to apply the r2_score along.
+    weights : ndarray
+        Input array of weights for a and b.
+    skipna : bool
+        If True, skip NaNs when computing function.
+
+    Returns
+    -------
+    res : ndarray
+        R^2 (coefficient of determination) score.
+
+    See Also
+    --------
+    sklearn.metrics.r2_score
+    """
+    sumfunc, meanfunc = _get_numpy_funcs(skipna)
+    if skipna:
+        a, b, weights = _match_nans(a, b, weights)
```

Comment:
does this line do all the nan handling? its quite extensive in https://github.com/scikit-learn/scikit-learn/blob/95d4f0841/sklearn/metrics/_regression.py#L488

Reply:
Yep. I copied and pasted most from `pearson_r` and used the sklean code to verify our implementation. It also allows weights to be input so I could verify that as well